### PR TITLE
remove net5.0 dependency in AzureCloudService

### PR DESCRIPTION
**Pull request by IsaacCalligeros95**
Upgrading net5.0 references to net6.0

### CODE REVIEW

**Comment by mjhilton** [comment]
Can you please add a bit of info about the extra code changes that were made @IsaacCalligeros95? Interested in the changes in warning numbers and certificate class.

File: source/Sashimi/CertificateEncoder.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ static void GrantCurrentUserAccessToPrivateKeyDirectory(string privateKeyPath)
             var folderPath = Path.GetDirectoryName(privateKeyPath);
             if (folderPath == null)
                 throw new Exception("There was no directory specified in the private key path.");
-#pragma warning disable PC001
+#pragma warning disable CA1416
```
Comment (IsaacCalligeros95) [inline source/Sashimi/CertificateEncoder.cs:144]: When upgrading to net6.0 the following code fails to compile due to treatWarningsAsErrors with CA1416 warnings, (only supported on 'windows)'. This is a language feature that was introduced with net5.0. A past [example PR](https://github.com/OctopusDeploy/Shellfish/commit/6ef5cd85af52710fa36433cf0aa30bc2cc27c438) of this pragma warning being added.

In the past the default assumption was that code would be [compatible on all platforms](https://docs.microsoft.com/en-us/dotnet/core/compatibility/code-analysis/5.0/ca1416-platform-compatibility-analyzer), this hasn't changed for these libraries with the release of net6.0.

image (1)

File: source/Sashimi/CertificateEncoder.cs [hunk 2]
```diff
@@ -150,14 +150,14 @@ static void GrantCurrentUserAccessToPrivateKeyDirectory(string privateKeyPath)
...
-#pragma warning restore PC001
+#pragma warning restore CA1416
         }
 
         static bool HasPrivateKey(X509Certificate2 certificate2)
         {
             try
             {
-                return certificate2.HasPrivateKey && certificate2.PrivateKey != null;
+                return certificate2.HasPrivateKey && certificate2.GetRSAPrivateKey() != null;
```
Comment (IsaacCalligeros95) [inline source/Sashimi/CertificateEncoder.cs]: .PrivateKey has been deprecated in favour of [GetRSAPrivateKey()](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.rsacertificateextensions.getrsaprivatekey?view=net-6.0) and both return a nullable if the cert doesn't have a valid private key. Given the [exceptions on .PrivateKey](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2.privatekey?view=net-6.0#System_Security_Cryptography_X509Certificates_X509Certificate2_PrivateKey) i.e. the key has to be an RSA or DSA key I think it's safe to just check for these two private key types. 

image

Comment (IsaacCalligeros95) [reply source/Sashimi/CertificateEncoder.cs]: I was missing the DSA key type previously and just added that. Keen to get your thoughts on this.

Comment (mjhilton) [reply source/Sashimi/CertificateEncoder.cs]: Yeah, I don't know 100% what needs to be supported here, but it feels that if we're migrating from the deprecated property that used to support both, we should check both. As long as those extension methods don't throw exceptions if the Private Key doesn't exist...?

**Review by mjhilton — APPROVED** [review]
LGTM

**Pull request merged**